pymc3/tests/test_stats.py

```diff
@@ -70,9 +70,11 @@ def test_compare():
     traces = [trace0] * 2
     models = [model0] * 2
```

Comment:
The `optimize` error pop-ups only if the compared models are the same, this was not a problem before this PR. Tomorrow I will check this in more detail.

Reply:
I just re-checked and I can confirm that the only problem is that models can't have the same name now. As in general it does not make sense to compare the same model (except maybe for debugging or testing) I guess the proper solution is to change the test and replace line 71 `models = [model0] * 2` with something like, 

```python
import copy
models = [model0, copy.copy(model0)]
```

Reply:
Those models would still have the same name, wouldn't they?

Reply:
You are right, There are a _couple_ of separate points here:

* Models are not allow to have the same name, that is correct.

* In this particular case models does not have names, and hence `compare` will assign the names automatically.

* For this test to work what really matters is that the models are different objects, otherwise you will get a dictionary with just one key:value pair.

Reply:
@fonnesbeck any update about this?

Reply:
I've implemented your suggestion. Thanks.